unison-cli/src/Unison/Codebase/Editor/Propagate.hs

```diff
@@ -393,20 +393,17 @@ propagate patch b = case validatePatch patch of
                       constructorReplacements',
                   seen'
                 )
-            doTerm :: Reference -> Cli (Maybe (Edits Symbol), Set Reference)
+            doTerm :: Reference -> Sqlite.Transaction (Maybe (Edits Symbol), Set Reference)
             doTerm r = do
               when debugMode (traceM $ "Rewriting term: " <> show r)
-              (componentMap, seen', mayComponent) <-
-                Cli.runTransaction do
-                  componentMap <- unhashTermComponent codebase r
-                  let componentMap' =
-                        over
-                          _2
-                          (Term.updateDependencies termReplacements typeReplacements)
-                          <$> componentMap
-                      seen' = seen <> Set.fromList (view _1 <$> Map.elems componentMap)
-                  mayComponent <- verifyTermComponent codebase componentMap' es
-                  pure (componentMap, seen', mayComponent)
+              componentMap <- unhashTermComponent codebase r
+              let componentMap' =
+                    over
+                      _2
+                      (Term.updateDependencies termReplacements typeReplacements)
+                      <$> componentMap
```

Comment:
Do you want this here, or local to the definition of `mayComponent`?

Reply:
I think its scope could be pushed down into a let, but I'm not sure it matters much. It was originally just inline like this (before a recent PR that indented due to a `runTransaction` call, and I can see that the corresponding `doType` above also just defines it inline.

Reply:
Fixed